File: Data Structures & Algorithms/longest-consecutive-sequence/submission-12.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = set(nums)
        longest = 0

        for num in nums:
            if num - 1 not in res:
                length = 1
                while num + 1 in res:
                    length += 1
                    num += 1
                longest = max(longest, length)
        return longest

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-12.cs
public class Solution {
    public int FindMin(int[] nums) {
        int l = 0, r = nums.Length - 1;
        int res = nums[0];

        while (l <= r) {
            if (nums[l] <= nums[r]) {
                res = Math.Min(res, nums[l]);
                break;
            }

            int mid = l + ((r - l)/2);
            res = Math.Min(res, nums[mid]);
            if (nums[mid] >= nums[l]) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return res;
    }
}
